### DISCUSSION TITLE
How to build the peoject

### DISCUSSION THREAD

**Discussion by AlbertSnow** [trojan-gfw/igniter#371] [Q&A] [answered]
Can we make a document describing how to build the App?

I want to build the project from source code. But a lot of lack dependence issues.

For example:
 Lack file of libboost_program_options. A. to build libjni-helper. So. (Where I can find the file. )
Lack file of golibs. Aar (I already find that from other place)

My suggestion is :
Commit all dependdece file. At least have a default file.

By the way I use MAC to develop.

**Comment by wongsyrone** [comment, marked as answer]
The best way to solve all of your issues is to check the CI configuration file: https://github.com/trojan-gfw/igniter/blob/master/.circleci/config.yml

**Reply by AlbertSnow** [reply]
Thank you for the reply. Yes, the file can solve part problem, but not all of that.
For example, Where can I Find libboost_program_options file?

**Reply by AlbertSnow** [reply]
And, my suggestion is upload all of the real dependence library.
It can run directly. And you can explore the project, if users want to explore more.
Just like the word say, default is better than config.

**Reply by AlbertSnow** [reply]
Oh! I got what you mean. And the problem is already fixed.
Directly to run git command "git submodule update --init --recursive --remote".

**Reply by TchaikovDriver** [reply]
BTW, use `pull_golibs.py` at the root directory of the project to get the latest released `golibs.aar`.
